tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -42,6 +45,80 @@ const auto kAppId = "id";
 const auto kAppSecret = "secret";
 const auto kTestHrn = "hrn:here:data::olp-here-test:hereos-internal-test";
 
+std::string GenerateQuadTreeResponse(
+    olp::geo::TileKey root_tile, std::uint32_t depth,
+    const std::vector<std::uint32_t>& available_levels) {
+  std::vector<std::uint16_t> sub_quads;
+  std::vector<std::uint64_t> parent_quads;
+
+  // generate data
+  for (auto level : available_levels) {
+    if (level < root_tile.Level()) {
+      auto key = root_tile.ChangedLevelTo(level);
+      parent_quads.push_back(key.ToQuadKey64());
+    } else {
+      const auto level_depth = level - root_tile.Level();
+      if (level_depth > depth) {
+        continue;
+      }
+
+      const auto start_level_id = 1 << (level_depth * 2);
+      std::vector<std::uint64_t> layer_ids(start_level_id);
+      std::iota(layer_ids.begin(), layer_ids.end(), start_level_id);
```

Comment:
Are you sure this works? I don't think it works.
I would rather have that you work with `TileKey` and `QuadKey64Helper` to create the tree.

Reply:
yeah its just a power of 4 for here tiles.
Replaced with helper functions.